autopilot/hostscore.go

```diff
@@ -243,3 +276,93 @@ func randSelectByWeight(weights []float64) int {
 	}
 	return len(weights) - 1
 }
+
+// contractPriceForScore returns the contract price of the host used for
+// scoring. Since we don't know whether rhpv2 or rhpv3 are used, we return the
+// bigger one for a pesimistic score.
+func contractPriceForScore(h hostdb.Host) types.Currency {
+	cp := h.Settings.ContractPrice
+	if cp.Cmp(h.PriceTable.ContractPrice) > 0 {
+		cp = h.PriceTable.ContractPrice
+	}
+	return cp
+}
+
+func uploadCostForScore(cfg api.AutopilotConfig, h hostdb.Host, bytes uint64) types.Currency {
+	uploadSectorCostRHPv2, _ := rhpv2.RPCAppendCost(*h.Settings, cfg.Contracts.Period)
+
+	asc := h.PriceTable.AppendSectorCost(cfg.Contracts.Period)
+	uploadSectorCostRHPv3, _ := asc.Total()
+
+	numSectors := bytes / rhpv2.SectorSize
+	if bytes%rhpv2.SectorSize != 0 {
+		numSectors++
+	}
+	if uploadSectorCostRHPv2.Cmp(uploadSectorCostRHPv3) > 0 {
```

Comment:
shouldn't we check for overflows when multiplying with the amount of sectors to be safe

Reply:
this is after gouging so we should be fine 👍